### DISCUSSION TITLE
Transforming y and X on reduced forecaster

### DISCUSSION THREAD

**Discussion by ilkersigirci** [alan-turing-institute/sktime#2923] [Ideas]
In the recursive reduced sklearn/keras models, assume that I don't want to use any lagged values from y and provide X(which are not extracted from y) like in supervised learning. I think in this case reduction strategy shouldn't apply any transformation to y and X but this is not the case in the current sktime version. Both y and X shifted by `window_size`. I think this can be done if I were able to provide lag=0 to `Window_summarizer` method `transformers` parameter of `make_reduction`. Currently, its default is lag=1.

Actually, my question is why we need to transform the data if we provide X for reduced models?

**Comment by fkiraly** [comment]
@ilkersigirci, the problem comes from two different ways to use exogeneous data. I.e., are you using the exogeneous data with a time index that is contemporaneous with the prediction, or one that is contemporaneous with the time point you predict from?

The current reduction wrapper implements the latter, but the strategy you describe - "just predict `y` from `X` at the same time" - is a special case only of the former.

I've implemented a prototype re-implementation based on the `Lag` transformer here:
https://github.com/alan-turing-institute/sktime/pull/2833
which can do both, have a look. Also look at the docstring if you want mathematical details of how the two approaches of using exogeneous data differ.

At the developer days next week (are you attending?) we want to have a more detailed look at reduction, and potentially revisit it for a major re-work.

Would appreciate comments and feedback (in this thread, or in #2833, if specific about rework).

**Comment by ilkersigirci** [comment]
As you said, I would like to be able to use the former approach. I did see the implementation in  #2833, but I thought it is just for direct reduction strategy. For the dev days, I will try to attend it (first time actually :) ) to see the reduction reimplementation process.
